rclpy/rclpy/subscription.py

```diff
@@ -22,6 +22,26 @@
 MsgType = TypeVar('MsgType')
 
 
+class SubscriptionEventCallbacks:
+    """Container to provide middleware event callbacks for a Subscription."""
+
+    def __init__(
+        self,
+        deadline_callback: Callable = None,
```

Comment:
Python 3.7 `@dataclass` would be perfect here
https://docs.python.org/3/library/dataclasses.html

Reply:
I think we support Python3.5

Reply:
Yeah - found the official - Crystal suppored Python3.5 - I doubt they're bumping the version up for this release http://www.ros.org/reps/rep-2000.html#crystal-clemmys-december-2018-december-2019 - Ubuntu Xenial and Raspbian Stretch are both on 3.5 still